src/dune_rules/lib_flags.mli

```diff
@@ -24,7 +24,14 @@ module L : sig
 
   val to_iflags : Path.Set.t -> _ Command.Args.t
   val include_paths : ?project:Dune_project.t -> t -> Lib_mode.t -> Path.Set.t
-  val include_flags : ?project:Dune_project.t -> t -> Lib_mode.t -> _ Command.Args.t
+
+  val include_flags
+    :  ?project:Dune_project.t
+    -> ?direct:(Lib.t -> bool)
```

Comment:
Why can't we just pass a list: `([ Direct | Indirect] * Lib.t) list` as an argument? I'm not a fan of passing around callbacks.

Reply:
Hello,
I have removed the passed callback. Instead, now two lists are passed, one containing the direct dependencies (`requires_direct`) and the other contains the hidden dependencies (`requires_hidden`). The `requires_hidden` is also exposed in `Compilation_context.t`, and so can be used in other places when needed. 

The code have changed a little bit, and the diff in `lib_flags.ml` is now smaller.

Review appreciated!